### PR TITLE
Fixes bug of Terraform file name

### PR DESCRIPTION
**Pull request by shivamdurgbuns**
Terraform file name has changed since newer version and thus this fix will iterate over the names of file and pick the right one.

### CODE REVIEW

File: ocs_ci/utility/azure_utils.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 # default location of files with necessary azure cluster details
 SERVICE_PRINCIPAL_FILEPATH = os.path.expanduser("~/.azure/osServicePrincipal.json")
 TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:24]:
```suggestion
TERRRAFORM_FILENAME = "terraform.platform.auto.tfvars.json"
```

File: ocs_ci/utility/azure_utils.py [hunk 2]
```diff
@@ -13,15 +13,16 @@
...
 
 
 # default location of files with necessary azure cluster details
 SERVICE_PRINCIPAL_FILEPATH = os.path.expanduser("~/.azure/osServicePrincipal.json")
-TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
+TERRRAFORM_FILENAME = "terraform.platform.auto.tfvars.json"
+OLD_TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
 
 
 def load_cluster_resource_group(cluster_path, terraform_filename=TERRRAFORM_FILENAME):
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:28]: Let's set this to None here.
```suggestion
def load_cluster_resource_group(cluster_path, terraform_filename=None):
```

File: ocs_ci/utility/azure_utils.py [hunk 3]
```diff
@@ -39,7 +40,20 @@ def load_cluster_resource_group(cluster_path, terraform_filename=TERRRAFORM_FILE
     Returns:
         string with resource group name
     """
-    filepath = os.path.join(cluster_path, terraform_filename)
+    terraform_files = [
+        os.path.join(cluster_path, f)
+        for f in [OLD_TERRRAFORM_FILENAME, TERRRAFORM_FILENAME]
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:45]:
```suggestion
        for f in [terraform_filename, OLD_TERRRAFORM_FILENAME]
```

File: ocs_ci/utility/azure_utils.py [hunk 4]
```diff
@@ -39,7 +40,20 @@ def load_cluster_resource_group(cluster_path, terraform_filename=TERRRAFORM_FILE
     Returns:
         string with resource group name
     """
-    filepath = os.path.join(cluster_path, terraform_filename)
+    terraform_files = [
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:43]: I would also maybe do here:
if not terraform_filename:
and move rest of the code only to that block.

File: ocs_ci/utility/azure_utils.py [hunk 5]
```diff
@@ -39,7 +40,20 @@ def load_cluster_resource_group(cluster_path, terraform_filename=TERRRAFORM_FILE
...
+        os.path.join(cluster_path, f)
+        for f in [OLD_TERRRAFORM_FILENAME, TERRRAFORM_FILENAME]
+    ]
+    filepath = None
+    for tf_file in terraform_files:
+        if os.path.exists(tf_file):
+            filepath = os.path.join(cluster_path, tf_file)
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:50]: maybe we can directly use `terraform_filename` instead of `filepath`

And avoid line 47.

Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:50]: I think we can avoid cluster_path as argument of function and read it directly from config.

File: ocs_ci/utility/azure_utils.py (hunk 3 above)
Comment (shivamdurgbuns) [reply ocs_ci/utility/azure_utils.py:40]: updated!

File: ocs_ci/utility/azure_utils.py (hunk 2 above)
Comment (shivamdurgbuns) [reply ocs_ci/utility/azure_utils.py]: updated!

File: ocs_ci/utility/azure_utils.py (hunk 4 above)
Comment (shivamdurgbuns) [reply ocs_ci/utility/azure_utils.py:38]: updated!

File: ocs_ci/utility/azure_utils.py (hunk 5 above)
Comment (shivamdurgbuns) [reply ocs_ci/utility/azure_utils.py]: updated!

File: ocs_ci/utility/azure_utils.py [hunk 6]
```diff
@@ -22,31 +22,45 @@
 # default location of files with necessary azure cluster details
 SERVICE_PRINCIPAL_FILEPATH = os.path.expanduser("~/.azure/osServicePrincipal.json")
 TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
+OLD_TERRRAFORM_FILENAME = "terraform.azure.auto.tfvars.json"
 
 
-def load_cluster_resource_group(cluster_path, terraform_filename=TERRRAFORM_FILENAME):
+def load_cluster_resource_group(terraform_filename=None):
```
Comment (petr-balogh) [inline ocs_ci/utility/azure_utils.py:28]: We have discussed that we will delete this parameter as well. It's anyway not used now anyhow.

**Review by dahorak — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by clacroix12 — COMMENTED** [review]
Looks like there is a merge conflict that needs resolving. Has there been any verification run for this change?

**Comment by shivamdurgbuns** [comment]
Rebase to the current master, and there is no test written for now to verify this change, I am writing a test for this in another pr. I have verified it locally.

**Review by petr-balogh — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**